### PR TITLE
Bugfix: lesson completed component

### PR DESCRIPTION
**Pull request by webbyfox**
This changeset fixes the bug on the LessonCompeleted component. Initial `Post` and `Delete` request seems working fine but `Delete` request was failing silently hence subsequent `Post` request was raising 500

_Tick or delete as appropriate:_

### Workflow

 - [x] Ticket exists in Jira  https://uktrade.atlassian.net/browse/1067
 - [x] Jira ticket has the correct status.
 - [x] [Changelog](CHANGELOG.md) entry added.
 
### Merging

- [X] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

File: react-components/src/components/MarkLessonAsComplete/MarkLessonAsComplete.jsx [hunk 1]
```diff
@@ -22,8 +22,10 @@ const MarkLessonAsComplete = ({ endpoint }) => {
...
     } else if (persistedIsComplete !== isComplete) {
-      Services[isComplete ? 'setLessonComplete' : 'setLessonIncomplete'](endpoint)
-        .then(() => {
+      Services[isComplete ? 'setLessonComplete' : 'setLessonIncomplete'](
+        endpoint
+      )
+        .finally(() => {
           setPersistedIsComplete(isComplete)
         })
         .catch(() => {})
```
Comment (bobmeredith) [inline react-components/src/components/MarkLessonAsComplete/MarkLessonAsComplete.jsx:31]: I think you can lose  this catch as there is a 'finally' now.

**Review by bobmeredith — CHANGES_REQUESTED** [review]

File: react-components/src/components/MarkLessonAsComplete/MarkLessonAsComplete.jsx (hunk 1 above)
Comment (webbyfox) [reply react-components/src/components/MarkLessonAsComplete/MarkLessonAsComplete.jsx:31]: Sorry - I've deleted and rebase/reset brought it back. will do commit now.

Comment (webbyfox) [reply react-components/src/components/MarkLessonAsComplete/MarkLessonAsComplete.jsx]: Added

**Review by bobmeredith — APPROVED** [review]
LGTM

**Pull request merged**